### ISSUE TITLE
An error occurred when exporting the TensorRT file

### ISSUE THREAD

**Issue by zhangzhenyu-pony** [ultralytics/ultralytics#18057]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

(YOLOv11) dell@dell-PowerEdge-R940xa:~/zhangzy/yolov11-main$ python export.py
WARNING ?? Python>=3.10 is required, but Python==3.9.20 is currently installed 
WARNING ?? TensorRT requires GPU export, automatically assigning device=0
Ultralytics 8.3.13 ?? Python-3.9.20 torch-1.12.0+cu116 CUDA:0 (NVIDIA A100-PCIE-40GB, 40354MiB)
YOLOv11 summary (fused): 238 layers, 2,582,737 parameters, 0 gradients, 6.3 GFLOPs

PyTorch: starting from 'runs/detect/train2-SGD/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 7, 8400) (5.2 MB)
requirements: Ultralytics requirement ['onnxslim==0.1.34'] not found, attempting AutoUpdate...
requirements: ? AutoUpdate skipped (offline)

ONNX: starting export with onnx 1.17.0 opset 15...
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
ONNX: slimming with onnxslim 0.1.42...
ONNX: export success ? 5.3s, saved as 'runs/detect/train2-SGD/weights/best.onnx' (10.1 MB)

TensorRT: starting export with TensorRT 10.7.0...
[12/06/2024-11:06:41] [TRT] [W] Unable to determine GPU memory usage: CUDA driver version is insufficient for CUDA runtime version
[12/06/2024-11:06:41] [TRT] [W] Unable to determine GPU memory usage: CUDA driver version is insufficient for CUDA runtime version
[12/06/2024-11:06:41] [TRT] [I] [MemUsageChange] Init CUDA: CPU +0, GPU +0, now: CPU 3131, GPU 0 (MiB)
terminate called after throwing an instance of 'nvinfer1::APIUsageError'
  what():  CUDA initialization failure with error: 35. Please check your CUDA installation: http://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html
已放弃 (核心已转储)
<<I tried to export the trained best-pt file as a TensorRt-formatted.engine file that can run on Jetson Xavier NX. An error occurred while executing the export command, which was run on the server. Do I need to execute export commands on my Jetson device?>>

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zhangzhenyu-pony, thank you for your interest in Ultralytics 🚀! We suggest checking out our [Docs](https://docs.ultralytics.com) for helpful resources on export commands and troubleshooting. The [Export Tutorial](https://docs.ultralytics.com/modes/export/) section might be particularly useful for your use case.

It appears you're encountering an issue during TensorRT export. If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/)? This might include your `best.pt` model, exact export command, and a clear step-by-step description of your environment and workflow. This information would help us debug the issue faster.

### Initial Troubleshooting Suggestions
- Your logs indicate a Python version mismatch: `Python>=3.10` is required but only `Python-3.9.20` is installed. Please upgrade to Python 3.10 or later and try again.
- The error seems to highlight potential CUDA installation issues. Ensure your CUDA and TensorRT installations are compatible with your system's environment. Refer to NVIDIA's [installation guide](http://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html) for troubleshooting.
- TensorRT export typically runs better on the target device (e.g., Jetson Xavier NX). You may want to try executing the export command directly on the Jetson device to avoid compatibility mismatches.

### Upgrade
Ensure you are running the latest version of the Ultralytics package and dependencies in a compatible Python environment:

```bash
pip install -U ultralytics
```

Ensure all CUDA/[CuDNN](https://developer.nvidia.com/cudnn) and TensorRT installations are properly set up. System information and dependency details are critical for debugging export issues.

### Verified Environments
You can also test the export process in one of these verified environments with GPU support:
- **Colab**: <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Paperspace**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>

### Community Support
Feel free to join the Ultralytics community for discussions and additional support:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time help.
- [Discourse](https://community.ultralytics.com) for in-depth discussions.
- [Subreddit](https://reddit.com/r/Ultralytics) to share ideas and solutions.

An Ultralytics engineer will review your issue shortly to assist further. Thank you for your patience and for helping us make Ultralytics better! 😊

**Comment by zhangzhenyu-pony** [comment]
Thank you, I have solved the above problem, I downgraded the tensorrt version to match the CUDA version.

**Comment by Y-T-G** [comment]
TensorRT needs to be exported on the device you're going to use it.

Follow the Ultralytics Jetson guide

https://docs.ultralytics.com/guides/nvidia-jetson/#convert-model-to-tensorrt-and-run-inference
